tin/apps/submissions/models.py

```diff
@@ -239,12 +234,15 @@ def save_file(self, submission_text: str) -> None:
 
         fpath = self.file_path
 
-        os.makedirs(os.path.dirname(fpath), exist_ok=True)
+        if fpath is None:
+            raise ValueError("Cannot save file if file_path is not found")
```

Comment:
This line is also a bit concerning and should be tested.

Reply:
If it works right now, it shouldn't make a difference (however, I don't discourage some testing!). The problem occurs because the `file_path` can be `None` if `self.file` is `None`. I just added this because pyright was complaining about `fpath` being `None`.
Same with similar exceptions being raised below.

Reply:
Yep, I see that. Just want to make sure `fpath` is never actually `None` to begin with.